lib/rspec/support/spec/encoding_helpers.rb

```diff
@@ -0,0 +1,30 @@
+module RSpec
+  module Support
+    module EncodingHelpers
+      module_function
+      if String.method_defined?(:encoding)
+
+        def expect_identical_string(str1, str2, expected_encoding=str1.encoding)
+          str1.encoding == expected_encoding &&
+            str1.each_byte.to_a == str2.each_byte.to_a
+        end
+
+      else
+
+        def expect_identical_string(str1, str2)
+          str1.split(//) == str2.split(//)
+        end
+      end
+    end
+  end
+end
+require 'rspec/expectations'
+# Special matchers for comparing encoded strings so that
+# we don't run any expectation failures through the Differ,
+# which also relies on EncodedString. Instead, confirm the
+# strings have the same encoding and same bytes.
+RSpec::Matchers.define :be_identical_string do |expected|
+  match do |actual|
+    RSpec::Support::EncodingHelpers.expect_identical_string(actual, expected)
```

Comment:
> Also, if I add a test for this, how/where would I do that?

We definitely test matchers that ship with RSpec but for fairly simple custom matchers like this that we just use in our own specs we don't tend to spec them.  The "test" of them is that we see them pass and fail appropriately.

If you do want to test them, you can see the specs for our built-in matchers as a guide:

https://github.com/rspec/rspec-expectations/tree/master/lib/rspec/matchers/built_in

Basically, we test both positive and negative cases, passing and failing.  You can use `expect { expectation }.to raise_error(RSpec::Matchers::ExpectationNotMetError, /snippet of expected failure/)` (where `expectation` is a code expression that uses the matcher in a way you expect to fail) to test the failing case.

But as I said, I don't consider testing it necessary.